### PR TITLE
MODINV-491 add missing field

### PR DESCRIPTION
**Pull request by viktor-draban**
link to the ticket https://issues.folio.org/browse/MODINV-491

There was added inTransitDestinationServicePointId field in Item.json and based on it necessary changes to make possible update Items without lose that kind of information

### CODE REVIEW

File: src/main/java/org/folio/inventory/domain/items/Item.java [hunk 1]
```diff
@@ -107,6 +109,18 @@ public Item(String id,
     this.metadata = metadata;
   }
 
+  public Item(String id,String holdingId, String inTransitDestinationServicePointId, Status status,
+              String materialTypeId, String permanentLoanTypeId, JsonObject metadata) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/inventory/domain/items/Item.java:113]: Indenation

File: src/main/java/org/folio/inventory/domain/items/Item.java [hunk 2]
```diff
@@ -129,6 +143,13 @@ public Item withHrid(String hrid) {
     return this;
   }
 
+  public String getInTransitDestinationServicePointId() { return inTransitDestinationServicePointId; }
```
Comment (alexanderkurash) [inline src/main/java/org/folio/inventory/domain/items/Item.java:146]: New line symbols

File: src/main/java/org/folio/inventory/domain/items/Item.java [hunk 3]
```diff
@@ -473,7 +494,7 @@ public JsonObject getMetadata() {
 
   public Item copyWithNewId(String newId) {
     return new Item(newId,
-      holdingId, this.status, this.materialTypeId,
+      holdingId, inTransitDestinationServicePointId, this.status, this.materialTypeId,
```
Comment (alexanderkurash) [inline src/main/java/org/folio/inventory/domain/items/Item.java:497]: Indentation - natural

File: src/main/java/org/folio/inventory/domain/items/Item.java [hunk 4]
```diff
@@ -510,7 +531,7 @@ public Item copyWithNewId(String newId) {
 
   public Item changeStatus(ItemStatusName newStatus) {
     return new Item(this.id,
-      holdingId, new Status(newStatus), this.materialTypeId,
+      holdingId, inTransitDestinationServicePointId, new Status(newStatus), this.materialTypeId,
```
Comment (alexanderkurash) [inline src/main/java/org/folio/inventory/domain/items/Item.java:534]: Same here

File: src/main/java/org/folio/inventory/resources/ItemRepresentation.java [hunk 5]
```diff
@@ -112,6 +112,7 @@ private JsonObject toJson(
     includeIfPresent(representation, "title", instance, i -> i.getString("title"));
     includeIfPresent(representation, "callNumber", holding, h -> h.getString("callNumber"));
     includeIfPresent(representation, Item.HRID_KEY, item.getHrid());
+    includeIfPresent(representation, Item.TRANSIT_DESTINATION_SERVICE_POINT_ID_KEY, item.getInTransitDestinationServicePointId());
```
Comment (alexanderkurash) [inline src/main/java/org/folio/inventory/resources/ItemRepresentation.java:115]: 100 symbols

File: src/main/java/org/folio/inventory/support/ItemUtil.java [hunk 6]
```diff
@@ -140,6 +141,7 @@ public static JsonObject toStoredItemRepresentation(Item item) {
       itemToSend.put(Item.LAST_CHECK_IN, item.getLastCheckIn().toJson());
     }
     includeIfPresent(itemToSend, Item.HRID_KEY, item.getHrid());
+    includeIfPresent(itemToSend, Item.TRANSIT_DESTINATION_SERVICE_POINT_ID_KEY, item.getInTransitDestinationServicePointId());
```
Comment (alexanderkurash) [inline src/main/java/org/folio/inventory/support/ItemUtil.java:144]: 100

File: src/main/java/org/folio/inventory/support/ItemUtil.java [hunk 7]
```diff
@@ -279,6 +282,7 @@ public static JsonObject mapToJson(Item item) {
     }
 
     includeIfPresent(itemJson, Item.HRID_KEY, item.getHrid());
+    includeIfPresent(itemJson, Item.TRANSIT_DESTINATION_SERVICE_POINT_ID_KEY, item.getInTransitDestinationServicePointId());
```
Comment (alexanderkurash) [inline src/main/java/org/folio/inventory/support/ItemUtil.java:285]: 100

File: src/test/java/api/items/ItemApiExamples.java [hunk 8]
```diff
@@ -455,10 +458,12 @@ public void canUpdateExistingItem()
     newItemRequest = itemsClient.create(newItemRequest).getJson();
 
     assertThat(newItemRequest.getString("copyNumber"), is("cp"));
+    assertThat(newItemRequest.getString(Item.TRANSIT_DESTINATION_SERVICE_POINT_ID_KEY), notNullValue());
```
Comment (alexanderkurash) [inline src/test/java/api/items/ItemApiExamples.java:461]: Check value

File: src/test/java/api/items/ItemApiExamples.java [hunk 9]
```diff
@@ -425,6 +425,8 @@ public void canUpdateExistingItem()
     TimeoutException,
     ExecutionException {
 
+    String TRANSIT_DESTINATION_SERVICE_POINT_ID = "12115888-d7c8-54e7-8287-22e97f7250a4";
```
Comment (alexanderkurash) [inline src/test/java/api/items/ItemApiExamples.java:428]: -> variable, random()

**Review by alexanderkurash — APPROVED** [review]

File: src/main/java/org/folio/inventory/domain/items/Item.java (hunk 2 above)
Comment (viktor-draban) [reply src/main/java/org/folio/inventory/domain/items/Item.java]: done

File: src/main/java/org/folio/inventory/domain/items/Item.java (hunk 4 above)
Comment (viktor-draban) [reply src/main/java/org/folio/inventory/domain/items/Item.java]: done

File: src/main/java/org/folio/inventory/domain/items/Item.java (hunk 3 above)
Comment (viktor-draban) [reply src/main/java/org/folio/inventory/domain/items/Item.java]: done

File: src/main/java/org/folio/inventory/resources/ItemRepresentation.java (hunk 5 above)
Comment (viktor-draban) [reply src/main/java/org/folio/inventory/resources/ItemRepresentation.java]: done

File: src/main/java/org/folio/inventory/support/ItemUtil.java (hunk 7 above)
Comment (viktor-draban) [reply src/main/java/org/folio/inventory/support/ItemUtil.java]: done

File: src/test/java/api/items/ItemApiExamples.java (hunk 8 above)
Comment (viktor-draban) [reply src/test/java/api/items/ItemApiExamples.java]: done

File: src/test/java/api/items/ItemApiExamples.java (hunk 9 above)
Comment (viktor-draban) [reply src/test/java/api/items/ItemApiExamples.java]: done

**Pull request merged**